### PR TITLE
Added test case for "Error in ode solve"

### LINKED ISSUE #19831: error in ode solve
When I solve this system of ode 
```
from sympy import *
n = symbols('n', positive=True)
x0 = symbols('x_0')
t0 = symbols('t_0')
T = symbols('T')
x = Function('x')
y = Function('y')

eq = (Eq(Derivative(y(t), t),  x(t)),
      Eq(Derivative(x(t), t),  y(t) + 1))

sol = dsolve(eq, ics={y(t0): T, x(t0): x0})
print(sol)
```
I get the proper solution and everything is fine. But if I try 
```
from sympy import *

t = symbols('t')
n = symbols('n', positive=True)
x0 = symbols('x_0')
t0 = symbols('t_0')
T = symbols('T')
x = Function('x')
y = Function('y')

eq = (Eq(Derivative(y(t), t),  x(t)),
      Eq(Derivative(x(t), t), n*( y(t) + 1)))

sol = dsolve(eq, ics={y(t0): T, x(t0): x0})
print(sol)
```
I get the error
```
/usr/local/lib/python3.6/dist-packages/sympy/solvers/ode/ode.py in dsolve(eq, func, hint, simplify, ics, xi, eta, x0, n, **kwargs)
    606             else:
    607                 solvefunc = globals()['sysode_nonlinear_%(no_of_equation)seq_order%(order)s' % match]
--> 608             sols = solvefunc(match)
    609             if ics:
    610                 constants = Tuple(*sols).free_symbols - Tuple(*eq).free_symbols

/usr/local/lib/python3.6/dist-packages/sympy/solvers/ode/ode.py in sysode_linear_2eq_order1(match_)
   6606     if match_['type_of_equation'] == 'type7':
   6607         sol = _linear_2eq_order1_type7(x, y, t, r, eq)
-> 6608     return sol
   6609 
   6610 def _linear_2eq_order1_type1(x, y, t, r, eq):

UnboundLocalError: local variable 'sol' referenced before assignment
```
What surprises me is that when I manually simplify the second equation it can find the solution without errors
```
eq = (Eq(Derivative(y(t), t),  x(t)),
      Eq(Derivative(x(t), t), n* y(t) + n))

sol = dsolve(eq, ics={y(t0): T, x(t0): x0})
print(sol)
```

Sympy version is 
```
import sympy
print(sympy.__version__)
1.6.1
```

I didn't get this error on sympy 1.1.1
Please let me know if it is not an error but expected behavior and I have to manually simplify the equations.

### PR DESCRIPTION
**Pull request by Ayush-Malik**
#### References to other Issues or PRs

fixes #19831

#### Brief description of what is fixed or changed
Added a test case for issue number #19831

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Ayush-Malik** [comment]
@oscarbenjamin , I have maded the changes , My previous PR #19936 for this particular issue was closed by mistake , So i created this new PR 
Is it good to go 😃

File: sympy/solvers/ode/tests/test_systems.py [hunk 1]
```diff
@@ -1041,6 +1041,36 @@ def test_sysode_linear_neq_order1_type2():
...
+    x = Function('x')
+    y = Function('y')
+    T = symbols('T')
+
+    eq8 = (Eq(Derivative(y(t), t),  x(t)),
+    Eq(Derivative(x(t), t), n*( y(t) + 1)))
+    sol = [
+        Eq(y(t),
+        (-((-T*sqrt(n)*exp(sqrt(n)*t_0)/2 - n*Integral(exp(sqrt(n)*t_0), t_0)/2 + x_0*exp(sqrt(n)*t_0)/2)/sqrt(n) + Integral(n*exp(sqrt(n)*t)/2, t)/sqrt(n)))*exp(-sqrt(n)*t)
+        + (T*sqrt(n)*exp(-sqrt(n)*t_0)/2 - n*Integral(exp(-sqrt(n)*t_0), t_0)/2 + x_0*exp(-sqrt(n)*t_0)/2)*exp(sqrt(n)*t)/sqrt(n)
```
Comment (oscarbenjamin) [inline sympy/solvers/ode/tests/test_systems.py:1061]: These lines are too long.

Comment (Ayush-Malik) [reply sympy/solvers/ode/tests/test_systems.py]: Ok i will fix it 😀

Comment (Ayush-Malik) [reply sympy/solvers/ode/tests/test_systems.py]: @oscarbenjamin , is it good to go for now

**Comment by oscarbenjamin** [comment]
Okay, looks good.

Thanks!

**Pull request merged**